### PR TITLE
Admin, New Order, Search customer: As a super admin, search for customer within all enterprises

### LINKED ISSUE #9684: [BO] Can't find customers as superadmin
## Description

As superadmin, it's currently not possible to find customers by using the search box in /admin/orders/<order_nr>/customer. This works fine as the hub manager.

See #9683

## Steps to Reproduce

1. Login as super admin
2. Visit /orders
3. Click "New order"
4. Select an enterprise and order cycle
5. On the Customer Details tab try do find any customer by typing in the name or email -> no results
6. Logout and login as an entrerprise owner with customers
7. Repeat from 2. -> see that the query yields results

Alternatively run the spec #9683 on your local system.

## Animated Gif/Screenshot

image

## Workaround

?

## Severity

As per https://openfoodnetwork.slack.com/archives/C02TZ6X00/p1663530146433049
bug-s4: it's annoying, but you can use it

## Your Environment

* Version used: verified in v4.2.12
* Browser name and version: Firefox 104
* Operating System and version (desktop or mobile): Ubuntu 20.04

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Closes #9684 

#### What should we test?

1. Login as super admin
2. Visit /orders
3. Click "New order"
4. Select an enterprise and order cycle
5. On the Customer Details tab try do find any customer by typing in the name or email -> you should see results (many)
6. Logout and login as an entrerprise owner with customers
7. Repeat from 2. -> see that the query yields results, but only the one that the enterprise owner manage

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/controllers/spree/admin/search_controller.rb [hunk 1]
```diff
@@ -19,7 +19,12 @@ def known_users
 
       def customers
         @customers = []
-        if spree_current_user.enterprises.pluck(:id).include? search_params[:distributor_id].to_i
+        enterprise_ids = if spree_current_user.admin?
+                           Enterprise.pluck(:id)
```
Comment (mkllnk) [inline app/controllers/spree/admin/search_controller.rb]: Loading all enterprise ids just to check if one specific id is present seemed inefficient to me. I added a couple of commits to simplify this.

**Review by mkllnk — APPROVED** [review]

**Comment by mkllnk** [comment]
@jibees Can you review my new commits?

**Comment by jibees** [comment]
I can't approve my own PR, but thanks @mkllnk! Moving to Test Ready then.

**Comment by drummer83** [comment]
Hi @jibees,

cool, this is working as expected. 👍 

Just one note for documentation: The available customers in the search are the same for enterprise user and super admin user because the hub needs to be selected in the first step of order creation. Step 5 of 'What to test' above indicates that the super admin would have access to customers of ALL shops, but this is not the case. This is as expected and as before though.

Moving to ready to go! 🚀 

### Findings outside of this PR
- New order in backoffice: Customer search behaviour is different for initial search (waits for 3 characters and searches the beginning of email only) and follow-up search (also 2, 1 and 0 characters are searched for and also in the middle of email address)

**Pull request merged**